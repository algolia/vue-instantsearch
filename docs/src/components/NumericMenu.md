---
title: NumericMenu
mainTitle: Components
layout: main.pug
category: Components
withHeadings: true
navWeight: 6
editable: true
githubSource: docs/src/components/NumericMenu.md
---

The numeric menu list is a widget that displays a list of numeric filters in a list. Those numeric filters are pre-configured when creating the widget.

<a class="btn btn-static-theme" href="stories/?selectedKind=NumericMenu">ðŸ•¹ try out live</a>

## Usage

```html
<ais-numeric-menu
  attribute="price"
  :items="[
    { label: 'All' },
    { label: '<= 10$', end: 10 },
    { label: '10$ - 100$', start: 10, end: 100 },
    { label: '100$ - 500$', start: 100, end: 500 },
    { label: '>= 500$', start: 500 },
  ]"
/>
```

## Props

Name | Type | Default | Description | Required
---|---|---|---|---
attribute | String | - | The attribute | Yes
items | `Array<{ label: string, start?: number, end?: number }>` | - | Array of available options for the widget | Yes
transformItems | `(items: Array<{ label: string, value: string, isRefined: boolean }>)` | x => x | Function which receives the items, which will be called before displaying them. Should return a new array with the same shape as the original array. Useful for mapping over the items to transform, remove or reorder them |Â -
classNames | Object | | Override class names | no

## CSS classes

Here's a list of CSS classes exposed by this widget. To better understand the underlying DOM structure, have a look at the generated DOM in your browser.

Note that you can pass the prop `class-names`, with an object of class names and their replacement to override this.

Class name | Description
---|---
`.ais-NumericMenu` | the root div of the widget
`.ais-NumericMenu--noRefinement` | the root div of the widget with no refinement
`.ais-NumericMenu-list` | the list of all refinement items
`.ais-NumericMenu-item` | the refinement list item
`.ais-NumericMenu-item--selected` | the selected refinement list item
`.ais-NumericMenu-label` | the label of each refinement item
`.ais-NumericMenu-radio` | the radio input of each refinement item
`.ais-NumericMenu-labelText` | the label text of each refinement item
