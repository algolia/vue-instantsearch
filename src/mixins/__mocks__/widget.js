let state = {};

// we need to have state given by `component` before it is mounted, otherwise
// we can't render it in most cases (items, hits, etc. are used in the template)
// so we share a (mock) global state during a whole test.
//
// (a mock is imported once per test file, so the state is isolated between tests)
//
// This allows us to import this `__setState` function and call it in the test
// to give the necessary data before mounting.
export function __setState(newState) {
  state = newState;
}

export const createWidgetMixin = () => ({
  data() {
    return {
      state,
    };
  },
});
