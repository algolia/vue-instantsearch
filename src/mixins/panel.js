import Vue from 'vue';

export const PANEL_EMITTER_NAMESPACE = 'instantSearchPanelEmitter';
export const PANEL_CHANGE_EVENT = 'PANEL_CHANGE_EVENT';

export const createPanelProviderMixin = () => ({
  props: {
    emitter: {
      type: Object,
      required: false,
      default() {
        return new Vue({
          name: 'PanelProvider',
        });
      },
    },
  },
  provide() {
    return {
      [PANEL_EMITTER_NAMESPACE]: this.emitter,
    };
  },
  data() {
    return {
      canRefine: true,
    };
  },
  created() {
    this.emitter.$on(PANEL_CHANGE_EVENT, value => {
      this.updateCanRefine(value);
    });
  },
  beforeDestroy() {
    this.emitter.$destroy();
  },
  methods: {
    updateCanRefine(value) {
      this.canRefine = value;
    },
  },
});

export const createPanelConsumerMixin = ({ mapStateToCanRefine }) => ({
  inject: {
    emitter: {
      from: PANEL_EMITTER_NAMESPACE,
      default() {
        return {
          $emit: () => {},
        };
      },
    },
  },
  data() {
    return {
      state: null,
      hasAlreadyEmitted: false,
    };
  },
  watch: {
    state: {
      immediate: true,
      handler(nextState, previousState) {
        if (!nextState) {
          return;
        }

        const previousCanRefine = mapStateToCanRefine(previousState || {});
        const nextCanRefine = mapStateToCanRefine(nextState);

        if (!this.hasAlreadyEmitted || previousCanRefine !== nextCanRefine) {
          this.emitter.$emit(PANEL_CHANGE_EVENT, nextCanRefine);

          this.hasAlreadyEmitted = true;
        }
      },
    },
  },
});
