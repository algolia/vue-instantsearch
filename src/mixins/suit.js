import suit from '../util/suit';

export const createSuitMixin = ({ name }) => ({
  props: {
    classNames: {
      type: Object,
      default: undefined,
    },
  },
  methods: {
    suit(element, modifier) {
      const className = suit(name, element, modifier);
      const userClassName = this.classNames && this.classNames[className];
      if (userClassName) {
        return [className, userClassName].join(' ');
      }
      return className;
    },
  },
});
