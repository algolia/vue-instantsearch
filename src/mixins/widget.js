const defaultShouldUpdate = (previous, next) => previous !== next;

export const createWidgetMixin = ({ connect }) => ({
  inject: {
    instantSearchInstance: {
      name: 'instantSearchInstance',
      default() {
        const tag = this.$options._componentTag;
        throw new TypeError(
          `It looks like you forgot to wrap your Algolia search component
          "<${tag}>" inside of an "<ais-index>" component.`
        );
      },
    },
  },
  data() {
    return {
      state: null,
    };
  },
  computed: {
    widgetParams() {
      const { widgetParams } = this.$options;

      return Object.keys(widgetParams).reduce(
        (acc, name) => ({
          ...acc,
          [name]: widgetParams[name].getValue(this),
        }),
        {}
      );
    },
  },
  watch: {
    widgetParams(next, previous) {
      const { widgetParams } = this.$options;

      const shouldUpdateWidget = Object.keys(widgetParams).some(name => {
        const { shouldUpdate = defaultShouldUpdate } = widgetParams[name];

        return shouldUpdate(previous[name], next[name]);
      });

      if (shouldUpdateWidget) {
        this.instantSearchInstance.removeWidget(this.widget);
        this.widget = this.widgetFactory(next);
        this.instantSearchInstance.addWidget(this.widget);
      }
    },
  },
  created() {
    this.widgetFactory = connect(this.updateData, () => {});
    this.widget = this.widgetFactory(this.widgetParams);
    this.instantSearchInstance.addWidget(this.widget);
  },
  beforeDestroy() {
    this.instantSearchInstance.removeWidget(this.widget);
  },
  methods: {
    updateData(state = {}) {
      this.state = state;
    },
  },
});
