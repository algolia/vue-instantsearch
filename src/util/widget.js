export const addWidget = (connector, parent, widgetParams, render) => {
  const factory = connector(render);
  const widget = factory(widgetParams);
  parent.addWidgets([widget]);

  return widget;
};

export const forceRender = (widget, parent, instantSearchInstance) => {
  if (
    instantSearchInstance.__initialSearchResults &&
    !instantSearchInstance.started
  ) {
    if (typeof instantSearchInstance.__forceRender !== 'function') {
      throw new Error(
        'You are using server side rendering with <ais-instant-search> instead of <ais-instant-search-ssr>.'
      );
    }
    instantSearchInstance.__forceRender(widget, parent);
  }
};

export const removeWidget = (widget, parent) => {
  if (widget) {
    parent.removeWidgets([widget]);
  }
};
