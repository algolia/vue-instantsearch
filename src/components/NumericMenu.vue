<template>
  <div
    v-if="state"
    :class="[suit(), !canRefine && suit('', 'noRefinement')]"
  >
    <slot
      :items="state.items"
      :can-refine="canRefine"
      :refine="state.refine"
      :createURL="state.createURL"
    >
      <ul :class="[suit('list')]">
        <li
          v-for="item in state.items"
          :key="item.label"
          :class="[suit('item'), item.isRefined && suit('item', 'selected')]"
        >
          <label :class="suit('label')">
            <input
              type="radio"
              :class="suit('radio')"
              :name="attribute"
              :value="item.value"
              :checked="item.isRefined"
              @change="state.refine($event.target.value)"
            >
            <span :class="suit('labelText')">{{ item.label }}</span>
          </label>
        </li>
      </ul>
    </slot>
  </div>
</template>

<script>
import { connectNumericMenu } from 'instantsearch.js/es/connectors';
import { createPanelConsumerMixin } from '../mixins/panel';
import { createWidgetMixin } from '../mixins/widget';
import { createSuitMixin } from '../mixins/suit';

export default {
  name: 'AisNumericMenu',
  mixins: [
    createWidgetMixin({ connector: connectNumericMenu }),
    createSuitMixin({ name: 'NumericMenu' }),
    createPanelConsumerMixin({
      mapStateToCanRefine: state => !state.hasNoResults,
    }),
  ],
  props: {
    attribute: {
      type: String,
      required: true,
    },
    items: {
      type: Array,
      required: true,
    },
    transformItems: {
      type: Function,
      default(items) {
        return items;
      },
    },
  },
  computed: {
    widgetParams() {
      return {
        attribute: this.attribute,
        transformItems: this.transformItems,
        items: this.items,
      };
    },
    canRefine() {
      return !this.state.hasNoResults;
    },
  },
};
</script>
