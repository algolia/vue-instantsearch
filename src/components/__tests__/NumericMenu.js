import { mount } from '@vue/test-utils';
import { __setState } from '../../mixins/widget';
import NumericMenu from '../NumericMenu.vue';

jest.mock('../../mixins/widget');
jest.mock('../../mixins/panel');

const all = {
  label: 'All',
};

const lessThan10 = {
  label: '<= 10$',
  // Follow the InstantSearch implementation
  value: encodeURI(JSON.stringify({ end: 10 })),
  isRefined: false,
};

const from10to100 = {
  label: '10$ - 100$',
  // Follow the InstantSearch implementation
  value: encodeURI(JSON.stringify({ start: 10, end: 100 })),
  isRefined: false,
};

const from100to500 = {
  label: '100$ - 500$',
  // Follow the InstantSearch implementation
  value: encodeURI(JSON.stringify({ start: 100, end: 500 })),
  isRefined: false,
};

const moreThan500 = {
  label: '>= 500$',
  // Follow the InstantSearch implementation
  value: encodeURI(JSON.stringify({ start: 500 })),
  isRefined: false,
};

const defaultState = {
  items: [all, lessThan10, from10to100, from100to500, moreThan500],
  hasNoResults: false,
  createURL: () => {},
  refine: () => {},
};

const defaultProps = {
  attribute: 'brand',
  items: [
    { label: 'All' },
    { label: '<= 10$', end: 10 },
    { label: '10$ - 100$', start: 10, end: 100 },
    { label: '100$ - 500$', start: 100, end: 500 },
    { label: '>= 500$', start: 500 },
  ],
};

it('accepts an attribute prop', () => {
  __setState(defaultState);

  const props = {
    ...defaultProps,
  };

  const wrapper = mount(NumericMenu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.attribute).toBe('brand');
});

it('accepts an items prop', () => {
  __setState(defaultState);

  const props = {
    ...defaultProps,
  };

  const wrapper = mount(NumericMenu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.items).toEqual([
    { label: 'All' },
    { label: '<= 10$', end: 10 },
    { label: '10$ - 100$', start: 10, end: 100 },
    { label: '100$ - 500$', start: 100, end: 500 },
    { label: '>= 500$', start: 500 },
  ]);
});

it('accepts an transformItems prop', () => {
  __setState(defaultState);

  const identity = x => x;

  const props = {
    ...defaultProps,
    transformItems: identity,
  };

  const wrapper = mount(NumericMenu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.transformItems).toBe(identity);
});

describe('default render', () => {
  it('renders correctly', () => {
    __setState(defaultState);

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without refinement', () => {
    __setState({
      ...defaultState,
      hasNoResults: true,
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a selected value', () => {
    __setState({
      ...defaultState,
      items: [
        all,
        lessThan10,
        { ...from10to100, isRefined: true },
        from100to500,
        moreThan500,
      ],
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls refine on radio change', () => {
    const refine = jest.fn();

    __setState({
      ...defaultState,
      refine,
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
    });

    const input = wrapper.findAll('.ais-NumericMenu-radio').at(3);

    input.trigger('change');

    expect(refine).toHaveBeenCalledTimes(1);
    expect(refine).toHaveBeenCalledWith(expect.stringContaining('100'));
    expect(refine).toHaveBeenCalledWith(expect.stringContaining('500'));
  });
});

it('calls the Panel mixin with `hasNoResults`', () => {
  __setState({ ...defaultState });

  const wrapper = mount(NumericMenu, {
    propsData: defaultProps,
  });

  const mapStateToCanRefine = () =>
    wrapper.vm.mapStateToCanRefine(wrapper.vm.state);

  expect(mapStateToCanRefine()).toBe(true);

  wrapper.setData({
    state: {
      hasNoResults: true,
    },
  });

  expect(mapStateToCanRefine()).toBe(false);

  expect(wrapper.vm.mapStateToCanRefine({})).toBe(false);
});

describe('custom default render', () => {
  const defaultScopedSlot = `
    <ul
      slot-scope="{ items, canRefine, refine, createURL }"
      :class="[!canRefine && 'no-refinement']"
    >
      <li
        v-for="item in items"
        :key="item.label"
        :class="[item.isRefined && 'selected']"
      >
        <a
          :href="createURL(item.value)"
          @click.prevent="refine(item.value)"
        >
          {{ item.label }}
        </a>
      </li>
    </ul>
  `;

  it('renders correctly', () => {
    __setState(defaultState);

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without refinement', () => {
    __setState({
      ...defaultState,
      hasNoResults: true,
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a selected value', () => {
    __setState({
      ...defaultState,
      items: [
        all,
        lessThan10,
        { ...from10to100, isRefined: true },
        from100to500,
        moreThan500,
      ],
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with an URL for the href', () => {
    __setState({
      ...defaultState,
      createURL: (value = 'all') => `/price/${value}`,
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls refine on link click', () => {
    const refine = jest.fn();

    __setState({
      ...defaultState,
      refine,
    });

    const props = {
      ...defaultProps,
    };

    const wrapper = mount(NumericMenu, {
      propsData: props,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    const link = wrapper.findAll('a').at(3);

    link.trigger('click');

    expect(refine).toHaveBeenCalledTimes(1);
    expect(refine).toHaveBeenCalledWith(expect.stringContaining('100'));
    expect(refine).toHaveBeenCalledWith(expect.stringContaining('500'));
  });
});
