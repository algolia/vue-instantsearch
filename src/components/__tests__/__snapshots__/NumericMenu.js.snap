// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom default render renders correctly 1`] = `

<div class="ais-NumericMenu">
  <ul class>
    <li class>
      <a>
        All
      </a>
    </li>
    <li class>
      <a>
        &lt;= 10$
      </a>
    </li>
    <li class>
      <a>
        10$ - 100$
      </a>
    </li>
    <li class>
      <a>
        100$ - 500$
      </a>
    </li>
    <li class>
      <a>
        &gt;= 500$
      </a>
    </li>
  </ul>
</div>

`;

exports[`custom default render renders correctly with a selected value 1`] = `

<div class="ais-NumericMenu">
  <ul class>
    <li class>
      <a>
        All
      </a>
    </li>
    <li class>
      <a>
        &lt;= 10$
      </a>
    </li>
    <li class="selected">
      <a>
        10$ - 100$
      </a>
    </li>
    <li class>
      <a>
        100$ - 500$
      </a>
    </li>
    <li class>
      <a>
        &gt;= 500$
      </a>
    </li>
  </ul>
</div>

`;

exports[`custom default render renders correctly with an URL for the href 1`] = `

<div class="ais-NumericMenu">
  <ul class>
    <li class>
      <a href="/price/all">
        All
      </a>
    </li>
    <li class>
      <a href="/price/%7B%22end%22:10%7D">
        &lt;= 10$
      </a>
    </li>
    <li class>
      <a href="/price/%7B%22start%22:10,%22end%22:100%7D">
        10$ - 100$
      </a>
    </li>
    <li class>
      <a href="/price/%7B%22start%22:100,%22end%22:500%7D">
        100$ - 500$
      </a>
    </li>
    <li class>
      <a href="/price/%7B%22start%22:500%7D">
        &gt;= 500$
      </a>
    </li>
  </ul>
</div>

`;

exports[`custom default render renders correctly without refinement 1`] = `

<div class="ais-NumericMenu ais-NumericMenu--noRefinement">
  <ul class="no-refinement">
    <li class>
      <a>
        All
      </a>
    </li>
    <li class>
      <a>
        &lt;= 10$
      </a>
    </li>
    <li class>
      <a>
        10$ - 100$
      </a>
    </li>
    <li class>
      <a>
        100$ - 500$
      </a>
    </li>
    <li class>
      <a>
        &gt;= 500$
      </a>
    </li>
  </ul>
</div>

`;

exports[`default render renders correctly 1`] = `

<div class="ais-NumericMenu">
  <ul class="ais-NumericMenu-list">
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value
        >
        <span class="ais-NumericMenu-labelText">
          All
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22end%22:10%7D"
        >
        <span class="ais-NumericMenu-labelText">
          &lt;= 10$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:10,%22end%22:100%7D"
        >
        <span class="ais-NumericMenu-labelText">
          10$ - 100$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:100,%22end%22:500%7D"
        >
        <span class="ais-NumericMenu-labelText">
          100$ - 500$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:500%7D"
        >
        <span class="ais-NumericMenu-labelText">
          &gt;= 500$
        </span>
      </label>
    </li>
  </ul>
</div>

`;

exports[`default render renders correctly with a selected value 1`] = `

<div class="ais-NumericMenu">
  <ul class="ais-NumericMenu-list">
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value
        >
        <span class="ais-NumericMenu-labelText">
          All
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22end%22:10%7D"
        >
        <span class="ais-NumericMenu-labelText">
          &lt;= 10$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item ais-NumericMenu-item--selected">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:10,%22end%22:100%7D"
        >
        <span class="ais-NumericMenu-labelText">
          10$ - 100$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:100,%22end%22:500%7D"
        >
        <span class="ais-NumericMenu-labelText">
          100$ - 500$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:500%7D"
        >
        <span class="ais-NumericMenu-labelText">
          &gt;= 500$
        </span>
      </label>
    </li>
  </ul>
</div>

`;

exports[`default render renders correctly without refinement 1`] = `

<div class="ais-NumericMenu ais-NumericMenu--noRefinement">
  <ul class="ais-NumericMenu-list">
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value
        >
        <span class="ais-NumericMenu-labelText">
          All
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22end%22:10%7D"
        >
        <span class="ais-NumericMenu-labelText">
          &lt;= 10$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:10,%22end%22:100%7D"
        >
        <span class="ais-NumericMenu-labelText">
          10$ - 100$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:100,%22end%22:500%7D"
        >
        <span class="ais-NumericMenu-labelText">
          100$ - 500$
        </span>
      </label>
    </li>
    <li class="ais-NumericMenu-item">
      <label class="ais-NumericMenu-label">
        <input type="radio"
               name="brand"
               class="ais-NumericMenu-radio"
               value="%7B%22start%22:500%7D"
        >
        <span class="ais-NumericMenu-labelText">
          &gt;= 500$
        </span>
      </label>
    </li>
  </ul>
</div>

`;
